gokitmiddlewares/stalemiddleware/middleware.go

```diff
@@ -0,0 +1,39 @@
+package stalemiddleware
+
+import (
+	"context"
+	"time"
+
+	"github.com/arquivei/foundationkit/app"
+	"github.com/go-kit/kit/endpoint"
+)
+
+func New(c Config) endpoint.Middleware {
+	ch := make(chan struct{})
```

Comment:
Nit: a channel with no size will block a reader or a writer until both ends are filled. Adding a good size to this channel will increase the chances of it being non-blocking while keeping the same semantics.

There's a non-blocking solution, too. Instead of using channels and `time.After`, you can use the time of the last request in something like (pseudocode)

```
time.Sleep(unhealthyTimeLength)
if compare(lastKnownRequestTime, now) > unhealthyTimeLength then
    SetAsUnhealthy
end

repeat
```